include/circt/Dialect/FIRRTL/FIRRTLTypes.h

```diff
@@ -82,6 +82,10 @@ class FIRRTLType : public Type {
   // Convenience methods for accessing recursive type properties
   //===--------------------------------------------------------------------===//
 
+  /// Return true if this is a "passive" type - one that contains no "flip"
+  /// types recursively within itself.
+  bool isPassive() const { return getRecursiveTypeProperties().isPassive; }
```

Comment:
I'll open a new pr with just this change, but I figured it was perfectly valid based on [this comment](https://github.com/llvm/circt/blob/main/lib/Dialect/FIRRTL/FIRRTLTypes.cpp#L2342) that ref types are explicitly not passive

Reply:
It could be a method on `RefType` but if you already know you have a reftype you probably don't need to ask the question (and if that's not clear your code is probably doing the wrong thing).

The point is it doesn't generalize to non-hw types-- for example, Property types-- even if the current RTP approach requires all types put something in that bit, and while there's an answer for RefType's it's also rather misleading re:implication of flow of data through it and what (rw)probe's are.

Anyway if you'd like to help untangle this that'd be great, it's something I hope to push on soon-ish (tm).

Reply:
I see, I'll just remove it for now

Reply:
Removed and reworked to pass in a hardware type if present, which avoids unnecessary casting. I'm holding off on explicitly supporting property types until they are more baked/there is a Chisel representation of them.